server/container_create.go

```diff
@@ -554,24 +558,28 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID string,
 			specgen.AddProcessEnv(key, value)
 		}
 	}
-	for _, item := range containerInfo.Config.Config.Env {
-		parts := strings.SplitN(item, "=", 2)
-		if len(parts) != 2 {
-			return nil, fmt.Errorf("invalid env from image: %s", item)
-		}
+	if containerImageConfig != nil {
```

Comment:
Is this check and the one at 578 necessary?  I think the err check at 543 should fire in this case and if not, should we add this check there and just error out?